src/haddock/libs/libpdb.py

```diff
@@ -165,13 +164,15 @@ def swap_segid_chain(pdb_file_path: FilePath,
 
 
 def sanitize(
-        pdb_file_path: FilePathT,
-        overwrite: bool = True,
-        custom_topology: Optional[FilePath] = None) -> Union[FilePathT, Path]:
+    pdb_file_path: FilePathT,
+    overwrite: bool = True,
+    custom_topology: Optional[FilePath] = None,
+) -> Union[FilePathT, Path]:
     """Sanitize a PDB file."""
+    to_keep = _to_keep.copy()
```

Comment:
regarding shallow and deep copies, I think this one needs to be a deep copy of the list (https://stackoverflow.com/questions/43983907/python-list-copy-shallow-vs-deep-copy).
Also requires the import of copy.deepcopy (`from copy import deepcopy`)
```suggestion
    to_keep = deepcopy(_to_keep)
```